### ISSUE TITLE
Commit 29d9e31 creates memory handling problem under macOS

### ISSUE THREAD

**Issue by ManoloFLTK** [fltk/fltk#413]
It turns out commit 29d9e31c51 creates a problem for the macOS platform:
the processing of timeouts is no longer "under the umbrella of an NSAutoreleasePool"
so operations done by timeouts may fail to properly release memory.

That's visible for example with any FLTK app, typing cmd-+ to create a
temporary yellow window which is deleted by a timeout. The temp window's
memory never gets deallocated because this requires an NSAutoreleasePool
umbrella.

I propose the attached patch where (pseudo-code)

```
Fl::wait() {
  ** platform-independent stuff **
  Fl::system_driver()->wait();
}
```
is replaced by :

```
Fl::wait() {
  do_widget_deletion();
  Fl::system_driver()->wait();
}

Fl_XXX_System_Driver::wait() {
  create NSAutoreleasePool on macOS, nothing on other platforms
  Fl::platform_independent_wait() // does the platform-independent stuff
  do platform-specific stuff  // the previous content of Fl_XXX_System_Driver::wait()
  release NSAutoreleasePool on macOS
}
```

The patch contains also some temporary code to visualize allocation and deallocation of window objects.

[platform-independent-wait.patch.txt](https://github.com/fltk/fltk/files/8309507/platform-independent-wait.patch.txt)

**Comment by ManoloFLTK** [comment]
OK. I see there was a problem in platform-independent-wait.patch.txt that changes to the time_to_wait value were not propagated. This is fixed in the V2 attached here.

[platform-independent-waitV2.patch.txt](https://github.com/fltk/fltk/files/8309802/platform-independent-waitV2.patch.txt)

**Comment by Albrecht-S** [comment]
Yes, this was the main problem. Meanwhile I created my own patch version:
[platform-independent-wait_v3.patch.txt](https://github.com/fltk/fltk/files/8310179/platform-independent-wait_v3.patch.txt)

My version does not introduce a new method in class `Fl` which is a public class. I'm using standard virtual method overloads instead. It's like subclasses of widgets call their parent's handle() methods etc.: like in your new patch the derived class calls the platform independent method to recalculate the time:
```
double Fl_Unix_System_Driver::wait(double time_to_wait)
{
  time_to_wait = Fl_System_Driver::wait(time_to_wait);
```
The basics are the same (I modified your patch). I also documented the driver method which is only visible in the docs if you enable the driver documentation option. There's no need to exclude anything from doxygen docs as in your case.

My patch still includes your macOS debug statements.

Looking at the result we could generally move the first statement of
```
double Fl::wait(double time_to_wait) {
  // delete all widgets that were listed during callbacks
  do_widget_deletion();
```
(i.e. the deletion of widgets) to `double Fl_System_Driver::wait(double time_to_wait)` as well.

**Comment by ManoloFLTK** [comment]
I'm glad you propose to use an `Fl_System_Driver` method rather than a pseudo private `FL` method. That's what I wanted to do initially, but I had refrained to do it because you wrote earlier that drivers should contain only platform-specific code.

I've reworded a bit the doc of `double Fl_System_Driver::wait(double)` and committed the fix (f5628aa).
